src/transformers/modeling_utils.py

```diff
@@ -1049,6 +1113,10 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
         from_pipeline = kwargs.pop("_from_pipeline", None)
         from_auto_class = kwargs.pop("_from_auto", False)
         _fast_init = kwargs.pop("_fast_init", True)
+        torch_dtype = kwargs.pop("torch_dtype", None)
+        torch_dtype_auto_detect = kwargs.pop("torch_dtype_auto_detect", False)
```

Comment:
Around design, could we have `torch_dtype` accept string or torch.dtype values, then we can have an `"auto"` value and get rid of `torch_dtype_auto_detect`?

Reply:
That's smart! Will change to that.